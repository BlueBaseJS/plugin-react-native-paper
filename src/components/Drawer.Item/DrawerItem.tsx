import { Drawer } from 'react-native-paper';
import { DrawerItemProps } from '@bluebase/components';
import { componentMapper } from '@bluebase/component-mapper';

export const DrawerItem = componentMapper<DrawerItemProps>(Drawer.Item, {
	icon: ({ icon }: DrawerItemProps) => icon && icon.name,
	label: 'title',
}, { rest: true });

