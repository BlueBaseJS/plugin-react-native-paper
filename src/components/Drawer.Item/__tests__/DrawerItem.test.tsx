import { DrawerItem } from '../DrawerItem';
import React from 'react';
import { shallow } from 'enzyme';

describe('DrawerItem', () => {
	test('should forward title prop as label', () => {
		const DrawerItemTemp = DrawerItem as any;

		const component = shallow(<DrawerItemTemp title="Item 1" />);
		// expect(component).toMatchSnapshot();
		expect(component.props().label).toBe('Item 1');
	});

	test('should forward icon.name prop as icon', () => {
		const DrawerItemTemp = DrawerItem as any;

		const component = shallow(
			<DrawerItemTemp title="Item 1" icon={{ name: 'add-a-photo', size: 8, color: 'primary' }} />
		);
		expect(component).toMatchSnapshot();
		expect(component.props().label).toBe('Item 1');
	});
});
