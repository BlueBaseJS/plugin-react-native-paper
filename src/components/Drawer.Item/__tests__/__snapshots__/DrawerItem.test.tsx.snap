// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerItem should forward icon.name prop as icon 1`] = `
<withTheme(Drawer.Item)
  icon={[Function]}
  label="Item 1"
/>
`;
